homeassistant/components/melcloud/sensor.py

```diff
@@ -93,23 +101,23 @@ async def async_setup_entry(hass, entry, async_add_entities):
     mel_devices = hass.data[DOMAIN].get(entry.entry_id)
     async_add_entities(
         [
-            MelDeviceSensor(mel_device, measurement, definition)
-            for measurement, definition in ATA_SENSORS.items()
+            MelDeviceSensor(mel_device, measurement, metadata)
+            for measurement, metadata in ATA_SENSORS.items()
             for mel_device in mel_devices[DEVICE_TYPE_ATA]
-            if definition[ATTR_ENABLED_FN](mel_device)
+            if metadata.enabled(mel_device)
         ]
         + [
-            MelDeviceSensor(mel_device, measurement, definition)
-            for measurement, definition in ATW_SENSORS.items()
+            MelDeviceSensor(mel_device, measurement, metadata)
+            for measurement, metadata in ATW_SENSORS.items()
             for mel_device in mel_devices[DEVICE_TYPE_ATW]
-            if definition[ATTR_ENABLED_FN](mel_device)
+            if metadata.enabled(mel_device)
         ]
         + [
-            AtwZoneSensor(mel_device, zone, measurement, definition)
+            AtwZoneSensor(mel_device, zone, measurement, metadata)
             for mel_device in mel_devices[DEVICE_TYPE_ATW]
             for zone in mel_device.device.zones
-            for measurement, definition, in ATW_ZONE_SENSORS.items()
-            if definition[ATTR_ENABLED_FN](zone)
+            for measurement, metadata, in ATW_ZONE_SENSORS.items()
+            if metadata.enabled(mel_device)
```

Comment:
```suggestion
            if metadata.enabled(zone)
```
Not completely sure about it, but I think this should be `zone` instead. At least based on the previous code.